### PR TITLE
Add support for the translation of schemas into Welsh

### PR DESCRIPTION
**Pull request by berroar**
### What is the context of this PR?

Adds support for the translation of the health themed `PHM` survey into Welsh.

Scripts have been added to generate translation template `.pot` files, to generate a translated schema using a `.po` file,  and a script to check that the latest release of the `eq-translations` repository is being used. These scripts were previously used to translate census schemas and have been re-added to the repo.

Also added poetry to manage python dependencies, and updated Github actions so that a check is run on PR to ensure that translations templates are up to date.

### How to review

Github actions build should pass once [this translations PR is merged](https://github.com/ONSdigital/eq-translations/pull/83).

Check that the updated ReadMe makes sense.

Test the scripts added (you will need to create a dummy PHM schema and `.pot` and `.po` files).

### CODE REVIEW

**Comment by berroar** [comment]
[Update eq-questionnaire-schemas to support translations (3 SP)](https://trello.com/c/flEnuaTn/5444-update-eq-questionnaire-schemas-to-support-translations-3-sp)

File: README.md [hunk 1]
```diff
@@ -8,6 +8,34 @@ To do this run `make run-validator` to run the validator and then run `make vali
...
+** N.B. Currently only supporting translations for Health themed surveys (PHM).
+
+Before running any translations script it is important to check that the latest release of the eq-translations repo is being
+used. This can be done using:
+```angular2html
+make translations-check
+```
+
+To generate a translation template `.pot` file in order to translate a schema, use the following command. It will generate a template file containing all the strings to be translated:
+```angular2html
```
Comment (petechd) [inline README.md:22]: I don't think we need these `angular2html` tags.

File: README.md [hunk 2]
```diff
@@ -8,6 +8,34 @@ To do this run `make run-validator` to run the validator and then run `make vali
```
Comment (petechd) [inline README.md:8]: Should we add a mention about `test-translation-templates` from MAKEFILE?

File: scripts/extract_translation_templates.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    print_filename_results(f"{source_file}", contents_match)
+
+    return contents_match
+
+
+def build_schema_templates(output_dir):
+    filenames = []
+
+    for pattern in SCHEMAS_TO_EXTRACT:
+        filenames = glob(pattern)
```
Comment (MebinAbraham) [inline scripts/extract_translation_templates.py:71]: You need to extend filenames, this is only going to extract the last pattern.

File: README.md (hunk 2 above)
Comment (berroar) [reply README.md:8]: Added now 👍

**Review by petechd — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -8,6 +8,38 @@ To do this run `make run-validator` to run the validator and then run `make vali
...
+schema:
+```
+make translate-schemas
+```
+The translated schema will be added to the `/schemas/{THEME}/{LANGUAGE_CODE}` directory.
+
+To check that translations are up to date use the following command (This check will run automatically when a pull request is raised):
+```
+make test-translate-templates
+```
```
Comment (MebinAbraham) [inline README.md:41]: I would title these for clarity.
- Generating Translation Templates
- Generating Translated Schemas
- Testing Translation Templates

File: README.md [hunk 5]
```diff
@@ -8,6 +8,38 @@ To do this run `make run-validator` to run the validator and then run `make vali
 
 **Schema file names must use snake case to be compatible with runner**
 
+## Generating translated schemas
```
Comment (MebinAbraham) [inline README.md:11]: Since author will be using this repo, we need to add a section about how to set up and install the dependencies. So how to install python (pyenv), how to install poetry, and how to install the dependencies.

**Comment by rmccar** [comment]
Few of the files have no newline at end of file warnings in github, other than that looks good

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+version = "0.1.0"
+description = "Questionnaire Schemas Repository"
+authors = ["ONSDigital"]
+
+
+[tool.poetry.dependencies]
+python = "^3.10"
+coloredlogs = "*"
+black = "*"
+isort = "*"
```
Comment (MebinAbraham) [inline pyproject.toml:12]: These should be dev dependencies. Worth adding pylint.
Can you also make sure these are added to the Makefile and Actions.

Comment (berroar) [reply pyproject.toml]: I've added pylint (fixed the warnings raised) and a lint step for the actions 👍

File: .pylintrc [hunk 7]
```diff
@@ -0,0 +1,621 @@
...
+ignored-modules=
+
+# Python code to execute, usually for sys.path manipulation such as
+# pygtk.require().
+#init-hook=
+
+# Use multiple processes to speed up Pylint. Specifying 0 will auto-detect the
+# number of processors available to use, and will cap the count on Windows to
+# avoid hangs.
+jobs=1
```
Comment (MebinAbraham) [inline .pylintrc:71]:
```suggestion
jobs=0
```

File: README.md [hunk 8]
```diff
@@ -2,12 +2,66 @@
...
 
 To do this run `make run-validator` to run the validator and then run `make validate-schemas` to validate the schemas.
 
 **Schema file names must use snake case to be compatible with runner**
 
+## Translations
+
+** N.B. Currently only supporting translations for Health themed surveys (PHM).
+
+Ensure that the latest release of the eq-translations repository has been installed before running any of the following scripts (the release version is checked automatically when running translations, scripts will not run if the release is out of date.)
```
Comment (MebinAbraham) [inline README.md:35]: There is no action here. I would reword to:
```suggestion
The latest release of the eq-translations package is required to successfully run the translation scripts. The release version is checked automatically when running translations commands and the scripts will error if the release version does not match the installed version.
```

File: scripts/extract_translation_templates.py [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+            if not success:
+                logger.error(
+                    "Translation templates are not up to date. Run make translation-templates to fix this"
+                )
+                sys.exit(1)
+
+            logger.debug("Translation templates are up to date.")
+        sys.exit(0)
+
+    build_schema_templates(f"{os.getcwd()}/translations")
```
Comment (MebinAbraham) [inline scripts/extract_translation_templates.py:128]: This will error when `translations/` dir doesn't exist. 
Can be handled by:
```suggestion
    translations_dir = f"{os.getcwd()}/translations"
    pathlib.Path(translations_dir).mkdir(parents=True, exist_ok=True)
    build_schema_templates(translations_dir)
```

File: README.md [hunk 10]
```diff
@@ -2,12 +2,66 @@
...
+### Generating Translation Templates
+
+To generate a translation template `.pot` file in order to translate a schema, use the following command. It will generate a template file containing all the strings to be translated:
+```
+make translation-templates
+```
+
+### Generating Translated Schemas
+
+To translate a schema, a `.po` file for the schema will need to be added to the `translations/{THEME}/{LANGUAGE_CODE}` directory containing strings from the schema
```
Comment (MebinAbraham) [inline README.md:46]:
```suggestion
To translate a schema, a `.po` file for the schema will need to be added to the `translations/{SURVEY_TYPE}/{LANGUAGE_CODE}/` directory containing strings from the schema. For example, `translations/health/cy/`
```

File: README.md [hunk 11]
```diff
@@ -2,12 +2,66 @@
...
+and the matching translations. 
+
+The `.po` file needs to be named in the following format: `{SCHEMA_NAME}_{LANGUAGE_CODE}.po`
```
Comment (MebinAbraham) [inline README.md:49]: If it is in a language specific folder, then why does the filename need the language code?. I would drop the code that requires it, as you should already be in the context of a language folder.

File: README.md [hunk 12]
```diff
@@ -2,12 +2,66 @@
...
+
+Once this in place, the following command can be run in order to generate a translated
+schema:
+```
+make translate-schemas
+```
+The translated schema will be added to the `/schemas/{THEME}/{LANGUAGE_CODE}` directory.
```
Comment (MebinAbraham) [inline README.md:56]:
```suggestion
The translated schema will be added to the `/schemas/{SURVEY_TYPE}/{LANGUAGE_CODE}/` directory.
```

File: README.md (hunk 11 above)
Comment (berroar) [reply README.md]: Removed this 👍

File: README.md [hunk 13]
```diff
@@ -2,12 +2,77 @@
...
+
+### Install Poetry
+
+We use poetry to manage the dependencies in this repository, to install poetry use:
+
+```
+curl -sSL https://install.python-poetry.org | python3 -
+```
+
+Install the dependencies for this project using, add the option `--without dev` to ignore dev dependencies.
```
Comment (MebinAbraham) [inline README.md:29]:
```suggestion
Install the dependencies for this project. Add the option `--without dev` to ignore dev dependencies. 
```

File: README.md [hunk 14]
```diff
@@ -2,12 +2,77 @@
...
 A registry for questionnaire schemas for eq-questionnaire-runner.
 
+
+## Setup
+
+It is recommended that you use [Pyenv](https://github.com/pyenv/pyenv) to manage your Python installations.
+
+Use pyenv to install the desired version of Python:
+```
+pyenv install <version>
```
Comment (MebinAbraham) [inline README.md:12]:
```suggestion
pyenv install
```

**Review by MebinAbraham — COMMENTED** [review]
Can you add `scripts/__pycache__/` to the .gitignore file.

**Review by MebinAbraham — APPROVED** [review]
Apart from the README changes. LGTM.

Once merged, can you add the 2 extra GitHub action checks as a required check in the branch settings.
